File: mario.c
#include <cs50.h>
#include <stdio.h>

void brick_builder(int n);
void print_space(int n);
void print_brick(int n);

int main(void)
{
    int i = -1;
    while (i <= 0 || i > 8)
    {
        i = get_int ("Please enter a positive height value between 0-8: ");
    }
    //printf("Height: %i\n", i);
    brick_builder(i);
}

void brick_builder(int n){
    for(int i = 1; i <= n; i++)
    {
        print_space(n - i);
        print_brick(i);
        printf("  ");
        print_brick(i);
        printf("\n");
    }
}

void print_space(int n){
    for(int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

void print_brick(int n){
    for(int i = 0; i < n; i++)
    {
        printf("#");
    }
}

File: sentimental-mario-more/mario.py
i = "-1"
while (int(i) <= 0 or int(i) > 8):
    try:
        i = int(input("Please enter a positive height value between 0-8: "))
    except ValueError:
        continue

# print("Height: " + str(i))


def brick_builder(height: int):
    for i in range(1, height + 1):
        print_space(height - i)
        print_brick(int(i))
        print("  ", end='')
        print_brick(int(i))
        print()


def print_space(spaces: int):
    for i in range(spaces):
        print(" ", end='')


def print_brick(bricks: int):
    for i in range(bricks):
        print("#", end='')


brick_builder(int(i))
